### PR TITLE
DB-6728 Apache Ranger Integration (2.5)

### PR DESCRIPTION
**Pull request by jleach4**
This is the public side of the ranger implementation.  

Key Concepts:

AuthorizationProvider is now a factory vs. a singleton.
Splice Ambari Service now has its own config files for ranger if someone wants to configure it.
HDP 2.6.3 assembly now requires the ranger service and ranger plugin from the enterprise repo.

### CODE REVIEW

File: db-drda/src/main/java/com/splicemachine/db/impl/drda/DRDAConnThread.java [hunk 1]
```diff
@@ -1532,6 +1532,9 @@ private int getConnFromDatabaseName() throws DRDAProtocolException
 		}
             
 	 	try {
+			System.out.println("Remote --> " + ((InetSocketAddress)this.getSession().clientSocket.getRemoteSocketAddress()).getAddress().toString().replace("/",""));
```
Comment (snowspeeder) [inline db-drda/src/main/java/com/splicemachine/db/impl/drda/DRDAConnThread.java]: System.out.println?

**Comment by jleach4** [comment]
@snowspeeder Fixed the system.out inclusion.

File: splice_machine/src/main/java/com/splicemachine/derby/impl/db/SpliceDatabase.java [hunk 2]
```diff
@@ -173,6 +176,12 @@ public void unfreeze() throws SQLException{
     public void checkpoint() throws SQLException{
         throw new SQLException("Unsupported Exception");
     }
+/*
```
Comment (snowspeeder) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/db/SpliceDatabase.java]: remove if no more needed

File: splice_machine/src/main/java/com/splicemachine/derby/impl/db/SpliceDatabase.java [hunk 3]
```diff
@@ -96,6 +96,9 @@ public void boot(boolean create,Properties startParams) throws StandardException
 
         configureAuthentication();
 
+        // setup authorization
```
Comment (snowspeeder) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/db/SpliceDatabase.java]: remove

File: db-engine/src/main/resources/META-INF/services/com.splicemachine.db.iapi.services.authorization.AuthorizationFactory [hunk 4]
```diff
@@ -0,0 +1,15 @@
+#
+# Copyright (c) 2012 - 2017 Splice Machine, Inc.
```
Comment (snowspeeder) [inline db-engine/src/main/resources/META-INF/services/com.splicemachine.db.iapi.services.authorization.AuthorizationFactory]: minor: 2017 to 2018

File: db-engine/src/main/java/com/splicemachine/db/iapi/services/authorization/AuthorizationFactory.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
+package com.splicemachine.db.iapi.services.authorization;
```
Comment (snowspeeder) [inline db-engine/src/main/java/com/splicemachine/db/iapi/services/authorization/AuthorizationFactory.java]: do not see copyright notice

File: db-engine/src/main/java/com/splicemachine/db/iapi/services/authorization/AuthorizationFactory.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+import com.splicemachine.db.iapi.error.StandardException;
+import com.splicemachine.db.iapi.sql.conn.Authorizer;
+import com.splicemachine.db.iapi.sql.conn.LanguageConnectionContext;
+
+/**
+ *
+ *
```
Comment (snowspeeder) [inline db-engine/src/main/java/com/splicemachine/db/iapi/services/authorization/AuthorizationFactory.java]: empty comment block

**Pull request closed without merging**